### PR TITLE
hal/dx12: do not reverse the pipeline layout

### PR DESCRIPTION
**Pull request by kvark**
**Connections**
Fixes a validation error in DX12 that I found in one of the vange-rs code paths

**Description**
Since the idea of reversed layout came to the code, I fixed it twice (last time in #1744), and now it turns out there is not enough foundation for it to settle at all. Imagine the following scenario:
- BGLs: A, B, C, D
- PL[1]: A, B, C
- PL[2]: A, D

When switching from PL[1] to PL[2] wgpu-core doesn't try to re-bind `A`, since it's already bound, and it's preceeding any entries that changed. With reversed layout, however, `A` in these two pipeline layouts would occupy different slots. So it would have to be re-bound completely (as opposed to just marking the root signature elements as dirty).

So TL;DR: we can't do this right now because wgpu-core follows Vulkan pipeline compatibility rules. There is no principal reason for this: we could change the behavior based on the backend, but that's just the way it works now.

**Testing**
vange-rs

### CODE REVIEW

**Comment by kvark** [comment]
Included in wgpu-hal-0.10.2
